### PR TITLE
Add retry loop for when github api returns 202

### PR DESCRIPTION
**Pull request by benschwartz9**
[Github status codes](https://gist.github.com/subfuzion/669dfae1d1a27de83e69)

Sometimes Github will return a status code 202, meaning that the request was accepted but the data is not yet ready to deliver. Rather than error in this scenario, we can wait a while and repeat the request a certain number of times before giving up

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -115,16 +118,30 @@ func sendGithubRequestHelper(endpoint string, token string) (res *http.Response,
...
-		err = fmt.Errorf("Failed to send HTTP request")
-		statusCode = 500 // Internal server error
-	} else if res.StatusCode != 200 {
-		statusCode = res.StatusCode // forward API error code
-		err = fmt.Errorf("GitHub request responded with error code %d", statusCode)
+	for {
+		res, err = http.DefaultClient.Do(req)
+		retry_count += 1
+
+		if err != nil {
```
Comment (Ashassins) [inline api/api.go:127]: is status code ==202 counted as an error? maybe it err != nil should go last

**Review by Ashassins — CHANGES_REQUESTED** [review]
Thanks for doing this, I didn't think you'd try it so fast.

File: api/api.go (hunk 1 above)
Comment (benschwartz9) [reply api/api.go:127]: I don't think so? [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-successful-2xx) says this about 2xx status codes
> The 2xx (Successful) class of status code indicates that the client's request was successfully received, understood, and accepted.

Based on that definition, there would not be an error. Also, I tested this branch before making the PR. When there was a status code 202, this code was able to loop and retry the request. So, that means that `err` must have been nil when there is a 202.

Comment (Ashassins) [reply api/api.go:127]: sounds good

**Review by Ashassins — APPROVED** [review]

**Pull request merged**